### PR TITLE
Added index to README.md

### LINKED ISSUE #583: Add index to README.md
I'm thinking to add headers and simple explanations for the followings to the top page.
- Google Cloud Platform Java Dependency Dashboard
- Java Library Best Practices
- Linkage Checker Enforcer Rules
- Cloud OSS BOM

Current top page:

image

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #583 .

Markdown-formatted in readme branch: https://github.com/GoogleCloudPlatform/cloud-opensource-java/tree/readme

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,49 @@
 [![unstable](http://badges.github.io/stability-badges/dist/unstable.svg)](http://github.com/badges/stability-badges)
 
-This project explores common infrastructure and coding standards for open source Google Cloud Platform projects.
+This project explores common infrastructure, toolset and practices 
+to provide consistency among Google Cloud Java libraries. The consistency
+minimizes the risk of dependency conflicts caused by the libraries.
 
-[Code Hygiene Dashboard](https://storage.googleapis.com/cloud-opensource-java-dashboard/dashboard/target/dashboard/dashboard.html) (runs daily; work in progress)
+## Google Cloud Platform Java Dependency Dashboard
```
Comment (suztomo) [inline README.md]: This is not a h1 header to avoid being the same level as "# Development"

File: README.md [hunk 2]
```diff
@@ -1,24 +1,49 @@
...
+
+## Linkage Checker Enforcer Rule
+
+[Linkage Checker Enforcer Rule](./enforcer-rules) verifies that the transitive
+dependency tree of a Maven project or a [BOM](
+https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Management)
+does not have any [linkage errors](
+./library-best-practices/glossary.md#types-of-conflicts-and-compatibility).
+
+## Google Library BOM
```
Comment (suztomo) [inline README.md]: The name of the BOM needs to be changed after the artifact ID discussion.

File: README.md [hunk 3]
```diff
@@ -1,24 +1,49 @@
...
 
 # Development
 
 This project is built using _Maven_.
 
 ## Requirements
 
-1. The [Google Cloud SDK](https://cloud.google.com/sdk/); install
-  this somewhere on your file system.
+1. (Optional) The [Google Cloud SDK](https://cloud.google.com/sdk/); install
```
Comment (suztomo) [inline README.md]: Clarified this is optional. Only used for uploading the dashboard to GCS.

File: README.md [hunk 4]
```diff
@@ -1,24 +1,49 @@
...
+[Linkage Checker Enforcer Rule](./enforcer-rules) verifies that the transitive
+dependency tree of a Maven project or a [BOM](
+https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Management)
+does not have any [linkage errors](
+./library-best-practices/glossary.md#types-of-conflicts-and-compatibility).
+
+## Google Library BOM
+
+[Google Library BOM](boms/cloud-oss-bom) is a Bill-of-Materials (BOM) that holds
+list of Google Cloud Java libraries with their versions.
```
Comment (elharo) [inline README.md]: holds list --> lists

File: README.md [hunk 5]
```diff
@@ -1,24 +1,49 @@
...
+[The Java Library Best Practices](./library-best-practices) are a set of rules
+that we believe will minimize problems for consumers of interconnected Java
+libraries.
+
+## Linkage Checker Enforcer Rule
+
+[Linkage Checker Enforcer Rule](./enforcer-rules) verifies that the transitive
+dependency tree of a Maven project or a [BOM](
+https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Management)
+does not have any [linkage errors](
```
Comment (elharo) [inline README.md]: but it can have linkage errors

File: README.md [hunk 6]
```diff
@@ -1,24 +1,49 @@
...
+./library-best-practices/glossary.md#types-of-conflicts-and-compatibility).
+
+## Google Library BOM
+
+[Google Library BOM](boms/cloud-oss-bom) is a Bill-of-Materials (BOM) that holds
+list of Google Cloud Java libraries with their versions.
+We choose the managed dependencies so that they are consistent with each other
+as much as possible.
+For understanding how BOMs help library users avoid dependency conflicts, read
```
Comment (elharo) [inline README.md]: To understand

File: README.md [hunk 7]
```diff
@@ -1,24 +1,49 @@
...
+[Declaring Dependencies on Google Java Libraries](DECLARING_DEPENDENCIES.md)
 
 # Development
```
Comment (elharo) [inline README.md]: I don't see why this is H1 and the others H2

File: README.md (hunk 6 above)
Comment (suztomo) [reply README.md]: Fixed

File: README.md (hunk 4 above)
Comment (suztomo) [reply README.md]: Fixed

File: README.md (hunk 5 above)
Comment (suztomo) [reply README.md]: Removed "a BOM", to avoid BOM member ordering issue.

File: README.md [hunk 8]
```diff
@@ -1,24 +1,49 @@
 [![unstable](http://badges.github.io/stability-badges/dist/unstable.svg)](http://github.com/badges/stability-badges)
 
-This project explores common infrastructure and coding standards for open source Google Cloud Platform projects.
+This project explores tools and practices 
+to provide consistency among Google Cloud Java libraries. The consistency
+minimizes the risk of dependency conflicts caused by the libraries.
 
-[Code Hygiene Dashboard](https://storage.googleapis.com/cloud-opensource-java-dashboard/dashboard/target/dashboard/dashboard.html) (runs daily; work in progress)
+# Tools and Practices
```
Comment (elharo) [inline README.md]: This isn't needed. Just make each section an H1

File: README.md [hunk 9]
```diff
@@ -1,24 +1,49 @@
 [![unstable](http://badges.github.io/stability-badges/dist/unstable.svg)](http://github.com/badges/stability-badges)
 
-This project explores common infrastructure and coding standards for open source Google Cloud Platform projects.
+This project explores tools and practices
```
Comment (elharo) [inline README.md]: stet

File: README.md (hunk 7 above)
Comment (suztomo) [reply README.md]: I added '# Tools and Practices' as H1 header before these new H2.

File: README.md [hunk 10]
```diff
@@ -1,24 +1,49 @@
...
+
+## Java Library Best Practices
+
+[The Java Library Best Practices](./library-best-practices) are a set of rules
+that we believe will minimize problems for consumers of interconnected Java
+libraries.
+
+## Linkage Checker Enforcer Rule
+
+[Linkage Checker Enforcer Rule](./enforcer-rules) verifies that the transitive
```
Comment (elharo) [inline README.md]: "is a Maven enforcer rule that detects linkage errors in the current project"

File: README.md [hunk 11]
```diff
@@ -1,24 +1,49 @@
...
+dependency tree of a Maven project does not have any [linkage errors](
+./library-best-practices/glossary.md#types-of-conflicts-and-compatibility).
+
+## Google Library BOM
+
+[Google Library BOM](boms/cloud-oss-bom) is a Bill-of-Materials (BOM) that
+lists of Google Cloud Java libraries with their versions.
```
Comment (elharo) [inline README.md]: delete "of"
better yet, "provides consistent versions of Google Cloud Java libraries that work together without linkage errors"

File: README.md (hunk 9 above)
Comment (suztomo) [reply README.md]: Reverted, with 2 discussed modifications.

**Comment by suztomo** [comment]
@elharo Ping

File: README.md [hunk 12]
```diff
@@ -1,24 +1,46 @@
 [![unstable](http://badges.github.io/stability-badges/dist/unstable.svg)](http://github.com/badges/stability-badges)
 
-This project explores common infrastructure and coding standards for open source Google Cloud Platform projects.
+This project explores common infrastructure and best practices for open source
+Java projects for Google Cloud Platform.
```
Comment (suztomo) [inline README.md]: As discussed. Updated the sentence with "best practice" and "Java"

File: README.md [hunk 13]
```diff
@@ -1,37 +1,34 @@
 [![unstable](http://badges.github.io/stability-badges/dist/unstable.svg)](http://github.com/badges/stability-badges)
 
-This project explores tools and practices 
-to provide consistency among Google Cloud Java libraries. The consistency
-minimizes the risk of dependency conflicts caused by the libraries.
+This project explores common infrastructure and best practices for open source
+Java projects for Google Cloud Platform.
```
Comment (elharo) [inline README.md]: the Google Cloud Platform

Comment (suztomo) [reply README.md]: Updated.

**Pull request merged**